torch/distributions/multinomial.py

```diff
@@ -47,19 +47,19 @@ def __init__(self, total_count=1, probs=None, logits=None):
             raise NotImplementedError('inhomogeneous total_count is not supported')
         self.total_count = total_count
         self._categorical = Categorical(probs=probs, logits=logits)
-        batch_shape = probs.size()[:-1] if probs is not None else logits.size()[:-1]
-        event_shape = probs.size()[-1:] if probs is not None else logits.size()[-1:]
+        batch_shape = self._categorical.batch_shape
+        event_shape = self._categorical.probs_or_logits.size()[-1:]
         super(Multinomial, self).__init__(batch_shape, event_shape)
 
     @constraints.dependent_property
     def support(self):
         return constraints.integer_interval(0, self.total_count)
 
-    @property
+    @lazy_property
```

Comment:
Would this be cheaper as simply `.property`?

Reply:
Actually, both are the same for all practical purposes (ran a simple profiler), since the underlying implementation is lazy. Will simply revert to `@property`. cc. @alicanb